### DISCUSSION TITLE
Function Calling With OpenAI Assistants

### DISCUSSION THREAD

**Discussion by Jaikant** [Predictable-Dialogs/agent-embed#4] [General]
Function calling with OpenAI assistants can be invoked in two ways
- On the browser by passing a callback function to the embed code. 
- On the backend, which is currently close sourced but more secure if we need to pass credentials to any API calls.

I would be interested to know which would be your preferred way to invoke functions using OpenAI?

**Comment by Tessedhec** [comment]
thank you for this insight, actually it doesn't matter, the easiest way is to make it invisible to the user.

How do I set this up?
I want to retrieve an email with a webhook, is that possible?
Thanks

**Reply by Tessedhec** [reply]
In fact, by creating a scheme, I can do it on a custom GPT but not with PredictDialog, how can I do it?

**Reply by Jaikant** [reply]
> thank you for this insight, actually it doesn't matter, the easiest way is to make it invisible to the user.
> 
> How do I set this up? I want to retrieve an email with a webhook, is that possible? Thanks

The  fastest way to support this is if we expose a callback function on the frontend embed code within which you could trigger your webhook.
PRs are welcome!

**Reply by Tessedhec** [reply]
Thank you for your responsiveness. Will this soon be available on your platform?

**Reply by Jaikant** [reply]
Yes, I would be working on it.

**Reply by Jaikant** [reply]
> > thank you for this insight, actually it doesn't matter, the easiest way is to make it invisible to the user.
> > How do I set this up? I want to retrieve an email with a webhook, is that possible? Thanks
> 
> The fastest way to support this is if we expose a callback function on the frontend embed code within which you could trigger your webhook. PRs are welcome!

@artulloss I am planning to implement this feature on the backend since we don't want to expose the API credentials on the browser.
